### ISSUE TITLE
c7n-org run returns zero return code with failed run

### ISSUE THREAD

**Issue by johnt337** [capitalone/cloud-custodian#2656]
# c7n-org run returns zero return code with failed run

When executing a policy run against multiple accounts, failure to apply to accounts with access denied is reported as a zero return code.

## How to replicate this problem
1. pip install `c7n_org` (and if required `pip install --upgrade botocore`)
2. Generate the `accounts.yml` with steps from [here](https://github.com/capitalone/cloud-custodian/tree/master/tools/c7n_org#config-file-generation)
3. Run the command: `c7n-org run -c accounts.yml -s output -u test.yml --dryrun`
4. If you have the wrong policy role, some of the accounts will return `AccessDenied` similar to:
    ```bash
    $ c7n-org run -c ~/accounts.yml -s output -u test.yml --dryrun -t path:/Sandbox --region us-east-1
    2018-07-19 12:18:19,005: c7n_org:INFO Ran account:ate-io-product11 region:us-east-1 policy:s3-enable-versioning matched:1 time:0.01
    2018-07-19 12:18:19,011: c7n_org:INFO Ran account:ate-io-product7 region:us-east-1 policy:s3-enable-versioning matched:1 time:0.01
    2018-07-19 12:18:19,010: c7n_org:INFO Ran account:ate-io-product35 region:us-east-1 policy:s3-enable-versioning matched:2 time:0.00
    2018-07-19 12:18:19,959: c7n_org:WARNING Access denied account:ate-io-product3 region:us-east-1
    2018-07-19 12:18:19,961: c7n_org:WARNING Access denied account:ate-io-product2 region:us-east-1
    2018-07-19 12:18:19,970: c7n_org:WARNING Access denied account:ate-io-product6 region:us-east-1
    2018-07-19 12:18:19,972: c7n_org:WARNING Access denied account:ate-io-product1 region:us-east-1
    2018-07-19 12:18:19,996: c7n_org:WARNING Access denied account:ate-io-product5 region:us-east-1
    2018-07-19 12:18:20,016: c7n_org:WARNING Access denied account:ate-io-product12 region:us-east-1
    2018-07-19 12:18:20,865: c7n_org:WARNING Access denied account:ate-io-product4 region:us-east-1
    2018-07-19 12:18:20,938: c7n_org:WARNING Access denied account:ate-io-product8 region:us-east-1
    2018-07-19 12:18:21,239: c7n_org:WARNING Access denied account:ate-io-product10 region:us-east-1
    2018-07-19 12:18:25,407: c7n_org:WARNING Access denied account:ate-io-product9 region:us-east-1
    2018-07-19 12:18:25,432: c7n_org:INFO Policy resource counts Counter({'s3-enable-versioning': 4})
    $
    ```
5. Run the command: `echo $?` and it will return `0`
    ```bash
    $ echo $?
    $ 0
    ```

## Problem statement

The c7n-org command should return a non-zero return code when one of its child policy runs fails.
As it stands _Access denied_ messages are written to `STDERR`, all non-error messages to `STDOUT`, I can potentially read data from `STDERR` in a wrapper as a short-term work around, but it would be more reliable to exit non-zero on error.

Other relevant info:
- c7n-org version `0.3.1`
- Python version `2.7.15`
- OSX version (uname -v) `Darwin Kernel Version 17.6.0: Tue May  8 15:22:16 PDT 2018; root:xnu-4570.61.1~1/RELEASE_X86_64`
- Linux version `Linux 4.9.93-linuxkit-aufs #1 SMP Wed Jun 6 16:55:56 UTC 2018 x86_64 Linux`

## Proposed solution

Make c7n-org return non-zero if any account policy runs fail.

**Comment by johnt337** [comment]
This also holds true for an expired access token.

**Comment by johnt337** [comment]
In the short term, i'm using a wrapper script similar to:

```bash
RUNAS=${RUNAS:-default}
ACCOUNTS=${ACCOUNTS:-${HOME}/accounts.yml}
POLICY=${POLICY:-${HOME}/enable-s3-versioning.yml}
BASE=$PWD;
OUTPUT=${OUTPUT:-$BASE/output}

echo -n "running...";
aws-runas ${RUNAS} c7n-org run -c ${ACCOUNTS} \
    -s ${OUTPUT} \
    -u ${POLICY} \
    -t path:/Sandbox \
    --region us-east-1 \
    --dryrun \
    2> ${BASE}/STDERR 1> ${BASE}/STDOUT;
echo "done";

LINES=$(wc -l ${BASE}/STDERR | awk '{print$1}')
echo "${BASE}/STDERR: ${LINES} line(s)"
if [ ${LINES} -gt 0 ]; then
  ERRORS=$(grep -cE 'ERROR|WARNING' ${BASE}/STDERR)
  if [ ${ERRORS} -gt 0 ]; then
    echo "failed to run, found ${ERRORS} error(s)";
    cat ${BASE}/STDERR ${BASE}/STDOUT;
    exit 1;
  else
    echo "good run";
    cat ${BASE}/STDERR ${BASE}/STDOUT;
    exit 0;
  fi
fi
```

**Comment by kapilt** [comment]
sounds good, thanks for the detailed bug report.

**Comment by aardvarq** [comment]
@kapilt not sure where this ended up but it's causing me grief at the moment.
```
$ c7n-org run -c ../../config/accounts.yml \
> -a nonprod \
> -r us-east-1 \
> -s /var/log/c7n/ \
> -u c7n_tmp_west.yml ; printf "my exit status is $?"
2018-09-21 03:28:24,601: c7n_org:ERROR Exception running policy:asg-cloudwatch-resize-search-candidate-instances account:nonprod region:us-east-1 error:An error occurred (ValidationException) when calling the PutRule operation: Parameter ScheduleExpression is not valid.
2018-09-21 03:28:24,619: c7n_org:INFO Policy resource counts Counter()
my exit status is 0
```
I'm looking at `tools/c7n_org/c7n_org/cli.py` and wonder if we couldn't simply `raise SystemExit(1)`
here:
https://github.com/capitalone/cloud-custodian/blob/master/tools/c7n_org/c7n_org/cli.py#L489
and here:
https://github.com/capitalone/cloud-custodian/blob/master/tools/c7n_org/c7n_org/cli.py#L499

**Comment by aardvarq** [comment]
That update locally on my machine yields:
```
c7n-org run -c ../../config/accounts.yml -a nonprod -r us-east-1 -s /var/log/c7n/ -u c7n_tmp_west.yml ; printf "my exit sta>
2018-09-21 03:46:58,670: c7n_org:ERROR Exception running policy:asg-cloudwatch-resize-search-candidate-instances account:nonprod region:us-east-1 error:An error occurred (ValidationException) when calling the PutRule operation: Parameter ScheduleExpression is not valid.
2018-09-21 03:46:58,678: c7n_org:WARNING Error running policy in nonprod @ us-east-1 exception: 1
my exit status is 1

```

**Comment by aardvarq** [comment]
https://github.com/capitalone/cloud-custodian/pull/2911

**Comment by kapilt** [comment]
merged in #2659, fwiw, we don't simply raise system exit on first error, we finish running all policies, and then if there were any failures during policy exec, we then exit 1.
